### DISCUSSION TITLE
Can parameter values change the scope of data from initial sync (without Reinitializing)?

### DISCUSSION THREAD

**Discussion by needledrag** [Mimetis/Dotmim.Sync#724] [Q&A]
Hi all,

Quick question on Filters and adding a CustomWhere clause.
I'm testing sending a parameter from the client (SQLite) to the server SQL Server.  
The parameter is a comma delimted string which is used in the custom where caluse  -WHERE BrandId IN (@MyParam) of a specific table (DownloadOnly).

The values sent from the client can change if they change their preferred brands on the client side and they want to download the products in their new preferred brands.
I realise I'd have to take care of removing brands on the client side that they no longer prefer (have unchecked) on the client.
My first sync is fine, all the preferred brands are synced to the client.  
On subsequent calls (and the preferred brands change, ie. there may be new BrandId's in the parameter), the new records for the BrandId aren't synced to the client.

I can see the parameter value has the correct BrandId's in by viewing the parameter value in OnHttpGettingRequest.

I'm guessing that only new inserts on the server or changes to rows that you initially downloaded are going to be synced then? 
So your initial sync determines what data you're going to track (except for inserts that meet the criteria)? and changing the query parameter that requests data that falls outside of the data in your initial sync, requires a ReInitialize?  
Just making sure I'm not completely missing something.

Many thanks for your time and help.

**Comment by matevzcesar** [comment]
I think the problem is that the new brands are also being tracked, but you filtered them out in the first sync and no changes happened since then. 

This is a very simplified process, but essentially what's happening is:

1. Brands X and Y have a "change version" of 1.
2. You sync for the first time while filtering out only brand X. Since this is your firs sync you get the current version which is 1.
3. You change the filter to brand Y.
4. You request another sync, filtering all changes for brand Y later than "change version" 1. There are no changes since version 1. You get no data back. 

You can test this by doing a simple update statement on the new group of brands before syncing. You can set a column back to the same value for example. You should then get the changes.

That said, if "preferred brands" is a setting that changes regularly, you'd be better off by syncing all of them and filtering them out locally.
